### ISSUE TITLE
Returned path when referencing a non-existent property

### ISSUE THREAD

**Issue by jonasschmidt** [Cleidiano/warpath#84]
Ok, so this one is a bit more exotic, and it might be we were relying on unintentional behavior.

In the older version we were using, referencing a non-existent property after a filter expression would still yield a path:
```elixir
iex(2)> Warpath.query([%{"foo" => "bar"}], "$[?(@.foo=='bar')].new", result_type: :path)
{:ok, ["$[0]['new']"]}
```

Whereas on current `master` I get this:
```elixir
iex(3)> Warpath.query([%{"foo" => "bar"}], "$[?(@.foo=='bar')].new", result_type: :path)
{:ok, []}
```

It still works with an array when accessing the element by index like this though:
```elixir
iex(1)> Warpath.query([%{"foo" => "bar"}], "$[0].new", result_type: :path)
{:ok, "$[0]['new']"}
```

Is that a behavior we shouldn't be relying on anyway? It's a bit of a gray area because the path doesn't actually exist in the given JSON.

**Comment by cleidiano** [comment]
Since version 0.2.0 the warpath are trying to stay has closest with the initiative of standardize de json-path, since then some behaviors have been guided by the consensus produced by [json-path-comparison](https://cburgmer.github.io/json-path-comparison/).

Accessing a non-existent property in the old version produces the path because it was deviating from the consensus and it was fixed at version 0.3.0, but it seems that there is still something to do. 
Producing a path that doesn't exist should be considered a bug and should be fixed to keep the behavior consistent.

**Comment by jonasschmidt** [comment]
Ok, makes sense. I will fix our implementation accordingly to not rely on the broken behavior anymore. Thanks for clarifying.
